hayeon/programmers-1/약수의-개수와-덧셈.js

```diff
@@ -0,0 +1,19 @@
+function solution(left, right) {
+  let result = 0;
+
+  for (let i = left; i <= right; i++) {
+    let divisor = [];
+    for (let j = 0; j <= i; j++) {
+      if (i % j === 0) divisor.push(j);
+    }
+    result += divisor.length % 2 === 0 ? i : -i;
+  }
+  return result;
+}
```

Comment:
```suggestion
function solution(left, right) {
  let result = 0;

  for (let i = left; i <= right; i++) {
    const divisorArr = [];
    
    for (let j = 0; j <= i; j++) {
      if (i % j === 0) divisorArr.push(j);
    }
    result += divisor.length % 2 === 0 ? i : -i;
  }
  return result;
}
```
- 풀이 잘봤습니다 👍 단순 divisor보다는 배열이기에 `divisors, divisorArr`이 더 적절할 것 같습니다. 또한 배열이기에 let보다는 `const`로 변수 선언이 적절합니다 🙇‍♂️
- 변수 선언 시에 let과 const부분은 그냥 처음에 변수선언할 때 `1순위로 const를 먼저 선언해주세요` ! 그리고 재할당이 필요한 경우에만 let으로 수정해주세요 😄
- 또한, 변수 선언 부와 제어문의 `문단 분리`를 해주세요 :) 적절한 문단 분리가 좋은 가독성을 가져옵니다 👍
- 해당 문제를 배열로 관리해서 매번 푸쉬하고, 배열의 길이로 result에 값을 누적하는 방법도 있지만 단순히 `Count`로 관리하는 방법이 있습니다.
- 아래 풀이는 Count로 접근 한 풀이이며, 약수의 개수를 가져오는 함수를 분리시킨 풀이입니다 참고해주세요

<br />

```js
function solution(left, right) {
  let result = 0;

  for (let i = left; i <= right; i++) {
    result += getDivisorCount(i) % 2 === 0 ? i : -i;
  }

  return result;
}

function getDivisorCount(num) {
  let divisorCount = 0;

  for (let i = 1; i <= num; i++) {
    if (num % i === 0) divisorCount++;
  }

  return divisorCount;
}
```

Reply:
헉 저도 변수이름정할때 고민을 많이했는데 배열이니까 divisorArr 가 훨씬 낫겠네요... 그리고 어떻게 아셨는지 변수 선언할 때 let 할지 const 할지 고민 많이했는데ㅠㅠ 문단분리도 신경써서 해보겠습니다!! 감사합니닷! 밑에 풀이도 참고해서 다시 풀어보겠습니다!!